.cascade-code/Chapman.edu/test-section/tatwell/query_api_spike/query_api.vtl

```diff
@@ -0,0 +1,175 @@
+
+##
+## Hardcoded Params
+##
+
+
+##
+## Vars
+##
+## Cascade provides a $currentPage object to every format.
+
+## This can be selected when Page Block is present:
+##set ( $pageNode = $_XPathTool.selectSingleNode($contentRoot, "//system-page[@current = 'true']") )
+
+
+##
+## Macros
+##
+#macro ( debugCurrentPage )
+  <pre>
+  # currentPage
+  currentPage name: $currentPage.name
+  currentPage path: $currentPage.path
+  currentPage assetType: $currentPage.assetType
+  currentPage Hide from navigation?: $currentPage.metadata.getDynamicField('Hide from navigation').checkbox
+
+  # _PropertyTool.outputProperties( currentPage )
+  $_PropertyTool.outputProperties( $currentPage )
+  </pre>
+#end
+
+#macro ( debugPathTools )
+  #set ( $currentPath = $currentPage.path )
+  #set ( $pathList = $currentPath.split('/') )
+  #set ( $parentIndex = $pathList.size() - 2 )
+  #set ( $parentFolderName = $pathList.get($parentIndex) )
+  #set ( $grandParentPath = $_StringTool.substringBefore($currentPath, $parentFolderName) )
+
+  <pre>
+  # debugPathTools
+  currentPage.parentFolder: $currentPage.parentFolder
+  \- isNull: $_PropertyTool.isNull($currentPage.parentFolder)
+  currentPath: $currentPath
+  pathList.size: $pathList.size()
+  parentFolderName: $parentFolderName
+  grandParentPath: $grandParentPath
+  </pre>
+#end
+
+#macro ( queryCurrentPageParentSiblings )
+  ## Note: Velocity (Java) arrays don't have a pop or remove method.
+  #set ( $parentSiblings = [] )
+
+  #set ( $currentPath = $currentPage.path )
+  #set ( $pathList = $currentPath.split('/') )
+  #set ( $parentIndex = $pathList.size() - 2 )
+  #set ( $parentFolderName = $pathList.get($parentIndex) )
+
+  ## locateFolder fails with trailing slashes. Thus the leading slash added here.
+  ## This will also need some sanity checks for paths less than 3 folders deep.
+  #set ( $grandParentPath = $_StringTool.substringBefore($currentPath, "/${parentFolderName}") )
+  #set ( $grandParentFolder = $_.locateFolder($grandParentPath) )
+
+  #foreach( $parentSibling in $grandParentFolder.children )
+    ## add method returns boolean which will displayed if not assigned to var.
+    #set ( $discard = $parentSiblings.add($parentSibling) )
+  #end
+
+  <pre>
+  # queryCurrentPageParentSiblings
+  currentPath: $currentPath
+  grandParentPath: $grandParentPath
+  grandParentFolder isNull?: $_PropertyTool.isNull($grandParentFolder)
+  parentSiblings: ${_DisplayTool.list($parentSiblings,"/")}
+  grandParentFolder.path: $grandParentFolder.path
+
+  # ParentSiblings: title : visible?
+  #foreach( $parentSibling in $grandParentFolder.children )
+    #set ( $metaHide = $parentSibling.metadata.getDynamicField('Hide from navigation').value )
+    #set ( $isVisible = $metaHide != 'Yes' )
+    $parentSibling.name : $isVisible
+  #end
+  </pre>
+#end
+
+#macro ( queryUmbrellaCategory )
+  ## For off-canvas nav
+  ## This would normally need to be extracted from currentPage's path.
+  #set ( $umbrellaPath = "/law" )
```

Comment:
As noted, some path-grinding code with sanity checks is going to be needed here in real life. It will also require our hardcoded `$umbrellaCategories` list.